src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/ILGen.cs

```diff
@@ -919,31 +918,27 @@ private static void EmitNullableToNullableConversion(this ILGenerator il, Type t
             // construct result type
             ConstructorInfo ci = typeTo.GetConstructor(new Type[] { nnTypeTo });
             il.Emit(OpCodes.Newobj, ci);
-            il.Emit(OpCodes.Stloc, locTo);
```

Comment:
I could see how it could be possible for this meaning that the result would be in the local either way to be used in some sort of jitter optimisation, but that roslyn produces code closer to what this PR results in suggests I'm not getting in the way of any such opts behind the scenes.